src/platforms/posix/drivers/tonealrmsim/tone_alarm.cpp

```diff
@@ -851,6 +851,7 @@ tone_alarm_main(int argc, char *argv[])
 
 			if (buffer == nullptr) {
 				PX4_WARN("not enough memory memory");
+				fclose(fd);
 				return 1;
 			}
 
```

Comment:
It seems to me that, as `play_string` is called with `free_buffer=true` on line 863, there should be no need for an additional call to free.